Connecting_MonteCarlo_to_ModernAI/N2_2D_Ising_Model.ipynb

```diff
@@ -0,0 +1,1233 @@
+        "\n",
+        "    sqrt_n: The square root of the number of atoms in the lattice\n",
+        "    returns padded_lattice: A lattice of size (L+2, L+2)\n",
+        "    \"\"\"\n",
+        "\n",
+        "    # initialize the lattice using np.random.random for a lattice of size\n",
+        "    # LxL\n",
+        "\n",
+        "    new_lattice = \n",
+        "\n",
+        "    # create a lattice of zeros that has an extra row on the top and bottom,\n",
+        "    # and an extra column on the left and the right\n",
+        "    \n",
+        "    padded_lattice = \n",
+        "\n",
+        "    #mask lattice by setting values above 0.5 to 1, and everything else to -1\n",
+        "    new_lattice[new_lattice>0.5] =\n",
+        "    new_lattice[new_lattice!=1] = \n",
+        "\n",
+        "    # added step to create non-interacting atoms\n",
+        "    padded_lattice[1:sqrt_n+1, 1:sqrt_n+1] = \n",
+        "\n",
+        "    return np.array(padded_lattice)"
+      ],
+      "metadata": {
+        "id": "n8qQmWgRw-mu"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Complete the MCMC_step function below\n",
+        "\n",
+        "def MCMC_step(beta: float, lattice: np.array):\n",
+        "    \"\"\"\n",
+        "    Function to repeat the Monte Carlo Markov Chain for this system.  \n",
+        "    beta: the inverse temperature value for the MCMC step\n",
+        "    lattice: the system of spins that will be simulated\n",
+        "    returns: an updated version of the input lattice\n",
+        "    \"\"\"\n",
+        "    \n",
+        "    # Figure out the size of the lattice\n",
+        "    [rows, cols] = lattice.shape \n",
+        "\n",
+        "    # keep the neighbors inside the region\n",
+        "    for r in range(1,rows-1): \n",
+        "        for c in range(1,cols-1):\n",
+        "\n",
+        "            # sum over the nearest neighbors\n",
+        "            sum_NN =\n",
+        "\n",
+        "            # calculate the energy\n",
+        "            E_a = \n",
+        "            \n",
+        "            # re-calculate the energy for a spin state change\n",
+        "            E_b = -1*E_a\n",
+        "\n",
+        "            # choose whether to keep the new state or not\n",
+        "            if #<ENTER LOGIC STATEMENT HERE>\n",
+        "                lattice[r, c] *= -1\n",
+        "\n",
+        "    return lattice"
+      ],
+      "metadata": {
+        "id": "_DpvUfIGnCwr"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### 1b) Perform the following experiment \n",
+        "\n",
+        "By calling the `main_exercise1` function using different arguments:\n",
+        "\n",
+        "1. Re-run the simulation for lattice sizes of $\\sqrt{N}$ = [10, 15, 20, 25, 40].\n",
```

Comment:
For my laptop, it took 10 min of running just to get up to sqrt_n=25. Maybe you could drop the 40? Just so students aren't waiting around too long. Especially if they have a couple runs with mistakes.

Reply:
Thank you for your comment.  Since I was able to do a 40x40 system size in about 10 minutes using Google Colab, I will leave this here, but have added the following note for students in the exercise instructions:

>Note: This computation may take some time to complete. Discuss with your instructor how long you should wait for a simulation to run.

I will also update the Instructor solutions that students running this on their personal machines may need accommodations for different run times.